dspy/adapters/chat_adapter.py

```diff
@@ -2,16 +2,34 @@
 import json
 import re
 import textwrap
-from typing import get_args, get_origin
+from typing import Any, Dict, KeysView, List, Literal, NamedTuple, get_args, get_origin
 
 import pydantic
 from pydantic import TypeAdapter
+from pydantic.fields import FieldInfo
 
+from ..signatures.field import OutputField
+from ..signatures.signature import SignatureMeta
+from ..signatures.utils import get_dspy_field_type
 from .base import Adapter
 
 field_header_pattern = re.compile(r"\[\[ ## (\w+) ## \]\]")
 
 
+class FieldInfoWithName(NamedTuple):
+    """
+    A tuple containing a field name and its corresponding FieldInfo object.
+    """
+
+    name: str
+    info: FieldInfo
```

Comment:
Because `_format_field_value` now requires information about:

1. Whether the field is an input or output field
2. The field's type annotation,

we cannot simply pass string field names to `_format_field_value`. We need to pass a full `FieldInfo` object along with a field name. Accordingly, we define a class to represent both pieces of information.

Reply:
does FieldInfo not have the field name/prefix as a member already?

Reply:
It doesn't :/

Reply:
Ah well that would have been nice 😅